troubleshooting/devtools.md

```diff
@@ -0,0 +1,188 @@
+
+## Build and Run With the Debugger
+Go to the VS Code left icon bar and select the Run and Debug icon. At the top, click the dropdown button on the bar with the play symbol.  
+The following options will be presented
+- Fluent Bit (General Config)
+- Fluent Bit (CloudWatch)
+- Fluent Bit (S3)
+- Fluent Bit (Kinesis Streams)
+- Fluent Bit (Kinesis Firehose)
+- Fluent Bit (Go Plugins)
+- Fluent Bit -- No Build (General Config)
+- Fluent Bit -- (General Config with Valgrind)
+
+The first 5 options build Fluent Bit, and run Fluent Bit with the corresponding config file, found in `fluent-bit/.vscode/fluent-bit-config`. Fluent Bit will be run with the debugger.
+
+The next option "Go Plugins" will build Cloudwatch and Kinesis Streams Go Plugins found in the external-plugins folder and run the resulting `.so` lib files with Fluent Bit. Kinesis Firehose has not yet been setup to work with this launch configuration option.  Please feel free to submit a pr to add. Fluent Bit is run with the fluent-bit_go.conf file. Note: the c debugger does not break on Go code lines from the Go attached libraries.
+
+The following option "No Build" runs Fluent Bit without building it, which may be helpful if you are running Fluent Bit and don't want to wait for it to build.
+
+The last option (General Config with Valgrind) runs Fluent Bit with Valgrind.
+
+**Run Fluent Bit**  
+After selecting a launch option from the dropdown, press the green play button and wait for Fluent Bit to build and execute. Anywhere you with to inspect your code while executing, select the whitespace to the left of a line number to add a breakpoint which looks like a red stopsign. When this line of code is executed, the code will hault, and VS Code will switch into debug view. To the left in the Run and Debug panel, you will be able to see variables at the local scope, the call stack, various threads, and other helpful details.
+
+You can add variable expressions to watch in the left hand panel as well which is helpful when confirming the state of a few suspect variables.
+
+## Running Acutest
+Acutest (Another C Unit Test) is the system Fluent Bit uses for unit testing.
+
+Each Acutest test is built as a separate compilation target meaning that when CMake builds, we will have a different binary for each test, which appears in our `fluent-bit/build/bin` folder after building Fluent Bit.
+
+To run the unit tests in VSCode, first configure CMake
+```
+cmake -DFLB_DEV=On -DFLB_TESTS_RUNTIME=On -DFLB_TESTS_INTERNAL=On ../
+make
+```
+
+At the blue ribbon bar of VS Code, you should see a button that says Build [all]. Press that button.
+When complete, click the text to the right of the debug and play button, which allows us to select our debug target. Click the unit test you would like to run. Then press the debug button.
+
+If you insert breakpoints in you test you will notice that these breakpoints, even though we are running in debug mode ***will not be hit!!!***
```

Comment:
As it is an official doc, I think we might not want it to be too over. Just one mark is fine. like **will not be hit!***